logstash-core/lib/logstash/config/defaults.rb

```diff
@@ -6,6 +6,14 @@ module LogStash module Config module Defaults
 
   extend self
 
+  def input
```

Comment:
Well, I want to follow the convention established by the cpu worker strategy. I think its cleaner to have only methods than to pick and choose based on how dynamic stuff is.